neo/core/basesignal.py

```diff
@@ -91,10 +98,16 @@ def _rescale(self, signal, units=None):
                 raise ValueError("Units must be specified")
         elif isinstance(signal, pq.Quantity):
             # could improve this test, what if units is a string?
-            if units != signal.units:
+            # This test should be safe now
+            if pq.quantity.validate_dimensionality(units) != signal.dimensionality:
```

Comment:
You're right, it does not work for CompoundUnits causing signals with CompoundUnits to always be rescaled. This is fine because rescaling still does the right thing. So I'll for now change the comment because I don't know a better way to check this.